src/wolff_producers.erl

```diff
@@ -356,41 +363,39 @@ maybe_restart_producers(#{client_id := ClientId,
     end, Producers),
   St.
 
--spec cleanup_workers_table(wolff:name()) -> ok.
-cleanup_workers_table(Name) ->
-  Ms = ets:fun2ms(fun({{N, _Partition}, _Pid}) when N =:= Name -> true end),
+-spec cleanup_workers_table(wolff:client_id(), wolff:topic()) -> ok.
+cleanup_workers_table(ClientId, Topic) ->
+  Ms = ets:fun2ms(fun({{C, T, _}, _}) when C =:= ClientId andalso T =:= Topic -> true end),
   ets:select_delete(?WOLFF_PRODUCERS_GLOBAL_TABLE, Ms),
   ok.
 
-find_producer_by_partition(Name, Partition) ->
-  case ets:lookup(?WOLFF_PRODUCERS_GLOBAL_TABLE, {Name, Partition}) of
-    [{_, Pid}] ->
+find_producer_by_partition(ClientId, Topic, Partition) ->
```

Comment:
Just a curiosity: if `Partition = partition_count`, we get the count rather than the pid.  :joy_cat:

Reply:
this function is always called with a number unless when trying to find the number of partitions.

Reply:
add `is_integer(Partition)` guard.